src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -4,20 +4,26 @@ import com.fasterxml.jackson.core.`type`.TypeReference
 import com.fasterxml.jackson.module.scala.JsonScalaEnumeration
 import tech.cryptonomic.conseil.generic.chain.DataTypes.OperationType.OperationType
 import tech.cryptonomic.conseil.generic.chain.DataTypes.OrderDirection.OrderDirection
+import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
 
-import scala.util.Try
 
 /**
   * Classes used for deserializing query.
   */
 object DataTypes {
+  import io.scalaland.chimney.dsl._
 
   /** Default value of limit parameter */
   val defaultLimitValue: Int = 10000
 
   /** Max value of limit parameter */
   val maxLimitValue: Int = 100000
 
+  /** Trait representing query validation errors */
+  sealed trait QueryValidationError {
```

Comment:
extending Product with Serializable might be useful later on, for type inference

Reply:
sure, added